### PR TITLE
Jbeap 9995 branch 3.0

### PR DESCRIPTION
**Pull request by rsearls**

### CODE REVIEW

File: testsuite/integration-tests/src/test/java/org/jboss/resteasy/test/providers/atom/AtomComplexModelTest.java [hunk 1]
```diff
@@ -83,6 +82,13 @@
                 AtomComplexModelState.class,
                 AtomComplexModelUuid.class,
                 AtomComplexModelVersionNumber.class);
+
+        war.addAsManifestResource(PermissionUtil.createPermissionsXmlAsset(
```
Comment (kanovotn) [inline testsuite/integration-tests/src/test/java/org/jboss/resteasy/test/providers/atom/AtomComplexModelTest.java]: With Ingo's fix in this PR https://github.com/resteasy/Resteasy/pull/1126 (related to AbstractJAXBContextFinder and JAXBContextWrapper) it reveals subsequent issue with:
`at org.jboss.resteasy.plugins.providers.atom.Entry.getAnyOtherJAXBObject(Entry.java:366)
`
This should maybe get another doPrivilege block in the code, not permission in the testcase.

File: testsuite/integration-tests/src/test/java/org/jboss/resteasy/test/providers/atom/AtomProviderTest.java [hunk 2]
```diff
@@ -68,6 +71,14 @@
     public static Archive<?> deploy() {
         WebArchive war = TestUtil.prepareArchive(AtomProviderTest.class.getSimpleName());
         war.addClass(AtomProviderTest.class);
+
+        war.addAsManifestResource(PermissionUtil.createPermissionsXmlAsset(
```
Comment (kanovotn) [inline testsuite/integration-tests/src/test/java/org/jboss/resteasy/test/providers/atom/AtomProviderTest.java]: With https://github.com/resteasy/Resteasy/pull/1126 only permission
`new RuntimePermission("accessDeclaredMembers"))`
is required to make the test pass.

File: testsuite/integration-tests/src/test/java/org/jboss/resteasy/test/providers/jaxb/JaxbCacheTest.java [hunk 3]
```diff
@@ -39,9 +40,13 @@
         WebArchive war = TestUtil.prepareArchive(JaxbCacheTest.class.getSimpleName());
         war.addClass(JaxbCacheTest.class);
         // Arquillian in the deployment
-        war.addAsManifestResource(PermissionUtil.createPermissionsXmlAsset(new ReflectPermission("suppressAccessChecks"),
+        war.addAsManifestResource(PermissionUtil.createPermissionsXmlAsset(
```
Comment (kanovotn) [inline testsuite/integration-tests/src/test/java/org/jboss/resteasy/test/providers/jaxb/JaxbCacheTest.java]: With PR https://github.com/resteasy/Resteasy/pull/1126 no additional permissions are needed for this test.

**Comment by rsearls** [comment]
Revisions have been made, however not all the requested changes could be made.
-- InjectionTest
    SocketPermission is required.  A doPrivileged block can not be placed around  code in JaxrsInjectionTarget because it is an implementation of intf javax.enterprise.inject.spi.InjectionTarget and no exception is declared to be thrown by the method, i.e. there is no convenient way to pass the exception up the stack; it would not be good to eat the exception.

-- JaxbCacheTest
   Existing permissions still required

-- AtomComplexModelTest 
I don't find that adding a doPrivileged block in org/jboss/resteasy/plugins/providers/atom/Entry had the desired affect.  The existing testcase permissions were still required.

**Comment by kanovotn** [comment]
-- AtomComplexModelTest
You are right, application AtomComplexModelAtomAssetMetadtaProcessor still calls JAXBContext.newInstance() so the application itself needs getClassLoader permission.

**Comment by kanovotn** [comment]
@asoldano LGTM

**Pull request merged**